resources/aws_ec2.go

```diff
@@ -677,3 +681,40 @@ func (obj *AwsEc2Res) UnmarshalYAML(unmarshal func(interface{}) error) error {
 func (obj *AwsEc2Res) prependName() string {
 	return AwsPrefix + obj.GetName()
 }
+
+// snsServe returns an http server used to listen for sns messages.
+func (obj *AwsEc2Res) snsServe() *http.Server {
+	addr := obj.WatchListenAddr
+	// if addr is a port
+	if _, err := strconv.Atoi(obj.WatchListenAddr); err == nil {
+		addr = fmt.Sprintf(":%s", obj.WatchListenAddr)
+	}
+	srv := &http.Server{Addr: addr}
+	log.Printf("%s: Starting SNS Endpoint", obj)
+	http.HandleFunc("/", obj.postHandler)
+	log.Printf("%s: Listening on %s", obj, addr)
+	obj.wg.Add(1)
+	go func() {
+		defer obj.wg.Done()
+		if err := srv.ListenAndServe(); err != nil {
+			select {
+			case obj.awsChan <- &chanStruct{
+				err: errwrap.Wrapf(err, "error starting http server"),
+			}:
+			case <-obj.closeChan:
+			}
+			return
+		}
+	}()
+	return srv
+}
+
+// postHandler listens for posts on the SNS Endpoint.
+func (obj *AwsEc2Res) postHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method == "POST" {
+		post, _ := ioutil.ReadAll(req.Body)
+		log.Printf(string(post))
+	} else {
```

Comment:
can we flip this and have an early return?
eg req.Method != "POST" { err, return }